fetch-in-react-practice/client/src/components/Todos.js

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-unused-vars -- Remove me */
+import React, { useEffect, useState } from 'react';
+import PageTitle from './PageTitle';
+import TodoList from './TodoList';
+import TodoForm from './TodoForm';
+
+const url = (path) => `${process.env.REACT_APP_BASE_URL}${path}`;
+
+export default function Todos() {
+  const [todos, setTodos] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    const getJsonData = async () => {
+      try {
+        const response = await fetch(url('/api/todos'));
+        if(!response.ok) throw new Error(`Error Code: ${response.status} Error Message: It Broken`);
+        const dataJson = await response.json();
+        setTodos(dataJson);
+      } catch (err) {
+        setError(err);
+        console.error('Error:', err.message)
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    getJsonData();
+  }, [isLoading]);
+
+  async function addTodo(newTodo) {
+       try {
+         const response = await fetch(url('/api/todos'), {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json"
+          },
+          body: JSON.stringify(newTodo),
+      });
+
+       if (!response.ok) throw new Error(`Error Code: ${response.status} Error Message: It Broken`);
+        const postJson = await response.json();
+        const newTodoList = todos.concat(postJson);
+        setTodos(newTodoList);
+      } catch (err) {
+        setError(err);
+        console.error('Error:', err);
+      }
+  };
+
+  async function toggleCompleted(todoId) {
+   try {
+      let matchTodo = {};
```

Comment:
You declare this here but you don't use it until way down at line 58, and you only use it inside the `if` block. You should always declare variables as close to where you first use them as possible.